src/snowflake/snowpark/functions.py

```diff
@@ -6702,17 +6702,75 @@ def sproc(
         registered. Invoking :func:`sproc` with ``replace`` set to ``True`` will overwrite the
         previously registered function.
 
+        4. To describe the return type for a stored procedure that returns tabular data, use
+        one of the following ways:
+
+            - (Recommended) Describe the return type using :attr:`~snowflake.snowpark.types.StructType`
+              and :attr:`~snowflake.snowpark.types.StructField`. Set ``return_type =
+              StructType([StructField("a", DataTypeA()), ...])`` to describe the case
+              ``RETURNS TABLE(A DataTypeA, ...)``.
+
+            - Set ``return_type = StructType([])`` to describe the case ``RETURNS TABLE()``.
```

Comment:
what if return_type is not provided? Should it also imply RETURNS TABLE()?

Reply:
If no `return_type` is provided then sproc registration should fail right. We enforce return types in non-table return types as well.

Reply:
ah ok I misunderstood, return_type is needed to indicate it's a table stored proc. Yea I agree with Yijun that we should use StructType()